### PR TITLE
update s2s client for tts params pass through

### PR DESCRIPTION
**Pull request by junkin**

### CODE REVIEW

File: riva/clients/nmt/riva_streaming_s2s_client.cc [hunk 1]
```diff
@@ -162,12 +171,18 @@ main(int argc, char** argv)
     return 1;
   }
 
+  if (!FLAGS_tts_encoding.empty() && FLAGS_tts_encoding != "pcm" && FLAGS_tts_encoding != "opus") {
```
Comment (virajkarandikar) [inline riva/clients/nmt/riva_streaming_s2s_client.cc:174]: is formatting for the curly brace off here?

File: riva/clients/nmt/riva_streaming_s2s_client.cc [hunk 2]
```diff
@@ -70,6 +70,10 @@ DEFINE_bool(
     "True returns text exactly as it was said with no normalization.  False applies text inverse "
     "normalization");
 DEFINE_string(ssl_cert, "", "Path to SSL client certificatates file");
+DEFINE_string(tts_encoding, "", "TTS output encoding, currently either PCM or OPUS");
```
Comment (virajkarandikar) [inline riva/clients/nmt/riva_streaming_s2s_client.cc:73]: may be set default encoding to PCM ?

File: riva/clients/nmt/streaming_s2s_client.cc [hunk 3]
```diff
@@ -301,21 +326,33 @@ StreamingS2SClient::ReceiveResponses(std::shared_ptr<S2SClientCall> call, bool a
...
-    int16_t* audio_data = (int16_t*)audio.data();
-    size_t len = audio.length() / sizeof(int16_t);
-    std::copy(audio_data, audio_data + len, std::back_inserter(pcm_buffer));
-    if (audio.length() < TTS_CHUNK_SIZE_BYTES) { // last chunk for this request
-      ::riva::utils::wav::Write(
-          std::string("streaming_" + std::to_string(cur_response) + ".wav"), 44100,
-          pcm_buffer.data(), pcm_buffer.size());
-      cur_response++;
-      pcm_buffer.clear();
+    if (tts_encoding_.empty() || tts_encoding_ == "pcm") {
```
Comment (virajkarandikar) [inline riva/clients/nmt/streaming_s2s_client.cc:331]: can get rid of tts_encoding_.empty() check (here and few other places) as we check it at the beginning.

File: riva/clients/nmt/riva_streaming_s2s_client.cc (hunk 2 above)
Comment (rmittal-github) [reply riva/clients/nmt/riva_streaming_s2s_client.cc:73]: we are already setting it to pcm if the string is empty, so this case is taken care. Empty and pcm behave same.

File: riva/clients/nmt/riva_streaming_s2s_client.cc (hunk 1 above)
Comment (rmittal-github) [reply riva/clients/nmt/riva_streaming_s2s_client.cc:174]: looks correct, no?
curly brace is at the same line as if, other conditions in the same file also doing similarly (e.g., line 187)

File: riva/clients/nmt/streaming_s2s_client.cc (hunk 3 above)
Comment (rmittal-github) [reply riva/clients/nmt/streaming_s2s_client.cc:331]: empty is allowed, if the user doesn't pass the param. Actually, in the beginning, we are checking for error, i.e. if the encoding is not empty and anything other than pcm/opus are provided.

File: riva/clients/nmt/riva_streaming_s2s_client.cc (hunk 1 above)
Comment (virajkarandikar) [reply riva/clients/nmt/riva_streaming_s2s_client.cc:174]: Ok. yes. It looked odd due to page render.

File: riva/clients/nmt/riva_streaming_s2s_client.cc (hunk 2 above)
Comment (virajkarandikar) [reply riva/clients/nmt/riva_streaming_s2s_client.cc:73]: right. but if you set a default value here, you don't need to check for empty().
its fine either way.

**Review by rmittal-github — APPROVED** [review]

**Pull request merged**